fe/fe-core/src/test/java/com/starrocks/http/TableQueryPlanActionTest.java

```diff
@@ -182,4 +182,27 @@ public void testQueryPlanActionPruneEmpty() throws Exception {
             Assert.assertEquals("{\"partitions\":{},\"opaqued_query_plan\":\"\",\"status\":200}", respStr);
         }
     }
+
+    @Test
+    public void testQueryPlanActionWithInvalidWarehouse() throws Exception {
+        super.setUpWithCatalog();
+        RequestBody body =
+                RequestBody.create(JSON, "{ \"sql\" :  \" select k1 from " + DB_NAME + "." + TABLE_NAME + " \" }");
+        Request request = new Request.Builder()
+                .post(body)
+                .addHeader("Authorization", rootAuth)
+                .addHeader("warehouse", "invalid_warehouse")
+                .url(URI + PATH_URI)
+                .build();
+        try (Response response = networkClient.newCall(request).execute()) {
+            String respStr = Objects.requireNonNull(response.body()).string();
+            JSONObject jsonObject = new JSONObject(respStr);
+            System.out.println(respStr);
+            Assert.assertEquals(500, jsonObject.getInt("status"));
```

Comment:
this is not expected, FE should respond HTTP 400 BAD request.

Reply:
The exception comes from here, which all exception wiil be 500

https://github.com/StarRocks/starrocks/blob/6996187be517ba1b241e1eba1eccebe2ccf1bce1/fe/fe-core/src/main/java/com/starrocks/http/rest/TableQueryPlanAction.java#L221-L225

Reply:
should do input parameter check before execute the parameter, or do fine grain exception handling. It is the request's parameter invalidation, should not be considered as a HTTP 5XX error. Usually a HTTP 5XX error means the http client is not doing anything wrong, and can do endless retry without modification of the request as wish.

Reply:
@kevincai I changed the code to check and set warehouse before execute the query_plan, so that we can get the real error status and exception.
So this PR is for `TableQueryPlanAction`, which can solve the sr flink source connector warehouse issue.

For other http Action, i am not sure if need the warehouse. But we can also do similar check like this PR if other http Actions have any need.

Reply:
It is good enough to fix only in query_plan API in this PR.